src/components/DocsList/DocsList.js

```diff
@@ -0,0 +1,82 @@
+import { push } from "../../utils/router.js"
+import { escapeHTML } from "../../utils/escapeHTML.js"
+
+export default function DocsList({ $target, initialState, onCreate, onRemove }) {
+  const $docsList = document.createElement("div")
+  $docsList.classList.add("docs-list")
+  $target.appendChild($docsList)
+
+  this.state = initialState
+
+  this.setState = (nextState) => {
+    this.state = nextState
+    this.render()
+  }
+
+  this.getDocuments = async () => {
+    const docs = await request("/documents", {
+      method: "GET",
+    })
+    this.setState(docs)
+  }
+
+  const renderDocuments = (documents) => {
+    return documents
+      .map((doc) => {
+        const hasChildDocuments = doc.documents && doc.documents.length > 0
+        const childDocuments = hasChildDocuments ? renderDocuments(doc.documents) : ""
+        const documentTitle = escapeHTML(doc.title)
+
+        return `
+        <li data-id="${doc.id}" class="document-item">
+          <div class="document-item-container">
+            <div class="document-title-container">
+              <i class="fa-regular fa-file-lines"></i>
+              <p class="document-title">${documentTitle}</p>
+            </div>
+            <button class="create" type="button">
+              <i class="create fa-solid fa-plus"></i>
+            </button>
+            <button class="remove" type="button">
+              <i class="remove fa-regular fa-trash-can"></i>
+            </button>
+          </div>
+          ${hasChildDocuments ? `<ul class="documents-list">${childDocuments}</ul>` : ""}
+        </li>
+      `
+      })
+      .join("")
+  }
+
+  this.render = () => {
+    $docsList.innerHTML = `
+      <ul>
+        ${renderDocuments(this.state)}
+      </ul>
+    `
+  }
+
+  this.render()
+
+  $docsList.addEventListener("click", async (e) => {
+    const $li = e.target.closest(".document-item")
+
+    if ($li) {
+      const { id } = $li.dataset
+      const { classList, tagName } = e.target
+
+      if (classList.contains("create")) {
+        const newDocId = await onCreate(id)
+        push(`/documents/${newDocId}`)
+      } else if (classList.contains("remove")) {
```

Comment:
```suggestion
      } else if (classList.contains("remove") || e.target.closest(".remove")) {
```
remove 버튼을 누르다가 가끔씩 작동을 안해서 디버깅해보니, <svg> 태그 안의 <path>가 클릭 target으로 지정되는 경우는 인식을 못하더라고요, 위와같이 추가적인 체크를 해주니 잘 작동하는 것 같습니다!

Reply:
와 저는 계속 contains, closest, tagName ... 등등의 조건 써가면서 그나마 잘 동작하는 걸 넣었다고 생각했거든요.
그냥 `||` 넣어주면 되는건데 지금 보니까 왜 계속 조건을 하나로 넣었는지 의문이네요.
제가 오직 하나의 조건으로 엄청 깔끔하게 보이도록 작성하고 싶었나 봅니다..
감사합니다~